packages/smart-wallet/src/smartWallet.js

```diff
@@ -928,10 +929,14 @@ export const prepareSmartWallet = (baggage, shared) => {
                 const purseP = facets.helper.purseForBrand(b);
 
                 // Now send it back to the purse.
-                return E(purseP).deposit(p);
+                return E(purseP)
+                  .deposit(p)
+                  .then(amt => out.push(amt));
```

Comment:
Nit, we don't need the return value
```suggestion
                  .then(amt => { out.push(amt); });
```

Reply:
But as discussed below

```suggestion
                  .then(amt => {
                    out.push(amt);
                    brandPaymentRecord.delete(b);
                  });
```